### DISCUSSION TITLE
How to change strimzi cruise control default topic replication factor to 3?

### DISCUSSION THREAD

**Discussion by bharat15-ml** [strimzi/strimzi-kafka-operator#7717] [Q&A] [answered]
I need to increase the Strimzi cruise control default topic (modeltrainingsamples, partitionmetricsamples) replication factor to 3, tried it below configurations, but it doesn't work.

    apiVersion: kafka.strimzi.io/v1beta2
    kind: Kafka
    metadata:
      name: my-cluster
    spec:
      kafka:
        version: 3.3.1
        replicas: 3
        listeners:
        - name: plain
          port: 9092
          type: internal
          tls: false  
        - name: tls
          port: 9093
          type: internal
          tls: true
        config:
          offsets.topic.replication.factor: 3
          transaction.state.log.replication.factor: 3
          transaction.state.log.min.isr: 2
          default.replication.factor: 3
          min.insync.replicas: 2
          inter.broker.protocol.version: "3.3"
        storage:
          type: ephemeral
      zookeeper:
        replicas: 3
        storage:
          type: ephemeral
      entityOperator:
      topicOperator: {}
      userOperator: {}
      cruiseControl:
        config:
          cruise.control.modeltrainingsamples.topic: strimzi.cruisecontrol.modeltraingsamples   
          cruise.control.modeltraingsamples.topic.replication.factor: 3
          cruise.control.modeltrainingsamples.topic.min.insync.replicas: 2
          cruise.control.modeltrainingsamples.topic: strimzi.cruisecontrol.partitionmetricsamples
          cruise.control.partitionmetricsamples.topic.replication.factor: 3
          cruise.control.partitionmetricsamples.topic.min.insync.replicas: 2

Could you suggest changing the default topic replication factor of strimzi cruise control topics?

**Comment by scholzj** [comment]
So, does the cluster already exist? In that case, you might need to use the Kafka partition reassignment tool to do it. I would suspect these options to be used only when the topic is created.

**Reply by bharat15-ml** [reply]
Yes, the Kafka cluster i.e. "my-cluster" already exists, once the Kafka cluster was created then cruise control pods spun up.
I guess, These topics - "modeltrainingsamples", and "partitionmetricsample",  are related to cruise control deployments only. 
My question is just like Kafka we can provide custom configuration parameters in the config section, similarly can't we customize the cruise control config, which eventually creates cruise control default topics accordingly?

after the changes checking in "/tmp/strimzi.properties" inside Kafka pods but were not able to see explicit cruise control configurations for an above-specified topic. I suspect I am not passing it correctly but haven't found much info on here too - https://github.com/linkedin/cruise-control/wiki/Configurations

**Reply by kyguy** [reply]
Jakub is right, the options are only used when the topics are created, so even after updating the `Kafka` resource with configuration suggested below, the Cruise Control topics would need to be deleted for the changes to take effect

**Comment by kyguy** [comment, marked as answer]
For changing the default replication factor of the sample store topics [1], partition.metric.sample.store.topic and broker.metric.sample.store.topic, configure the Cruise Control config section of the Kafka resource like this:
```
apiVersion: kafka.strimzi.io/v1beta2
kind: Kafka
metadata:
  name: my-cluster
spec:
  ...
  cruiseControl:
    config: 
      sample.store.topic.replication.factor: 3
```
Note that the cruiseControl.config section of the Kafka resource can take most of the configurations listed in the Cruise Control wiki [2]. For more information checkout the Strimzi docs here [1] Put same answer on StackOverflow post as well [3]

[1] https://strimzi.io/docs/operators/latest/configuring.html#property-cruise-control-config-reference

[2] https://github.com/linkedin/cruise-control/wiki/Configurations#kafkasamplestore-configurations

[3] https://stackoverflow.com/questions/74596628/how-to-change-strimzi-cruise-control-default-topic-replication-factor-to-3/74607314#74607314

**Reply by bharat15-ml** [reply]
Got a catch here!! Thanks for your timely reply.
